### PR TITLE
Add the unit testing framework.

### PR DESCRIPTION
**Pull request by nomis52**

### CODE REVIEW

**Comment by nomis52** [comment]
Hmm, I can't assign more than one person. @joshua-moyerman  can you take a look as well please.

File: tests/include/TestUtils.h [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+#define ASSERT_GT(expected, output)  \
+  CPPUNIT_ASSERT((expected) > (output))
+
+#define ASSERT_GTE(expected, output)  \
+  CPPUNIT_ASSERT((expected) >= (output))
+
+#define ASSERT_DATA_EQUALS(expected, expected_length, actual, \
+                               actual_length)  \
+  ola::testing::_AssertDataEquals(\
```
Comment (peternewman) [inline tests/include/TestUtils.h]: Really?

File: tests/include/TestUtilsPrivate.h [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+#include <cppunit/SourceLine.h>
+#include <cppunit/extensions/HelperMacros.h>
+#include <set>
+#include <sstream>
+#include <vector>
+
+namespace testing {
+
+// Assert that two data blocks are the same.
+// Private, use ASSERT_DATA_EQUALS below.
```
Comment (peternewman) [inline tests/include/TestUtilsPrivate.h]: You mean in the other file?

File: tests/include/TestUtilsPrivate.h [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+void _AssertDataEquals(
+    const CPPUNIT_NS::SourceLine &source_line,
+    const uint8_t *expected,
+    unsigned int expected_length,
+    const uint8_t *actual,
+    unsigned int actual_length);
+
+// Private, use ASSERT_VECTOR_EQ below
```
Comment (peternewman) [inline tests/include/TestUtilsPrivate.h]: Again.

File: tests/lib/TestUtils.cpp [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+#include "TestUtils.h"
+
+#include <cppunit/extensions/HelperMacros.h>
+#include <string.h>
+#include <iostream>
+
+namespace testing {
+
+void _AssertDataEquals(
```
Comment (peternewman) [inline tests/lib/TestUtils.cpp]: Shouldn't this be in TestUtilsPrivate.cpp?

**Comment by peternewman** [comment]
LGTM.

File: tests/include/TestUtils.h (hunk 1 above)
Comment (nomis52) [inline tests/include/TestUtils.h]: Oopps. Fixed.

File: tests/include/TestUtilsPrivate.h (hunk 2 above)
Comment (nomis52) [inline tests/include/TestUtilsPrivate.h]: Correct. Fixed.

File: tests/lib/TestUtils.cpp (hunk 4 above)
Comment (nomis52) [inline tests/lib/TestUtils.cpp]: The 'Private' is just a convention to mean 'don't depend on this'. The definitions live in the file with the base name.

**Pull request merged**